pandas/io/excel/_base.py

```diff
@@ -101,10 +103,17 @@
     If io is not a buffer or path, this must be set to identify io.
     Supported engines: "xlrd", "openpyxl", "odf", "pyxlsb", default "xlrd".
     Engine compatibility :
-    - "xlrd" supports most old/new Excel file formats.
+
+    - "xlrd" supports most old/new Excel file formats but is no longer maintained
+      and not supported with Python >= 3.9.
     - "openpyxl" supports newer Excel file formats.
     - "odf" supports OpenDocument file formats (.odf, .ods, .odt).
     - "pyxlsb" supports Binary Excel files.
+
+    .. deprecated:: 1.2.0
+        The default value ``None`` is deprecated and will be changed to ``"openpyxl"``
```

Comment:
The default of None is only deprecated when you're reading an xls/xlsx file, not when reading any of the other formats like ods. 
It might be worth explaining here (outside of the deprecated box) what the default engine is if not specified (eg also that openpyxl is automatically used if xlrd is not present for xlsx files)